File: 30-days-of-code/day2/solution.cpp
#include <bits/stdc++.h>

using namespace std;

int calculate_total_cost(double meal_cost, int tip_percent, int tax_percent)
{
    double tip_cost = meal_cost * (tip_percent / 100.0);
    double tax_cost = meal_cost * (tax_percent / 100.0);
    return round(meal_cost + tip_cost + tax_cost);
}

int main()
{
    double meal_cost;
    int tip_percent;
    int tax_percent;
    cin >> meal_cost >> tip_percent >> tax_percent;

    int result = calculate_total_cost(meal_cost, tip_percent, tax_percent);
    cout << result << "\n";
}

File: 30-days-of-code/day2/solution.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
  inputString += inputStdin;
});

process.stdin.on('end', _ => {
  inputString = inputString.replace(/\s*$/, '')
    .split('\n')
    .map(str => str.replace(/\s*$/, ''));

  main();
});

function readLine() {
  return inputString[currentLine++];
}

function calculateTotalCost(mealCost, tipPercent, taxPercent) {
  const tipCost = mealCost * (tipPercent / 100.0);
  const taxCost = mealCost * (taxPercent / 100.0);
  return Math.round(mealCost + tipCost + taxCost);
}

function main() {
  const mealCost = parseFloat(readLine());
  const tipPercent = parseInt(readLine(), 10);
  const taxPercent = parseInt(readLine(), 10);

  const result = calculateTotalCost(mealCost, tipPercent, taxPercent);
  console.log(result);
}
